components/Input/SelectInputWithList/index.js

```diff
@@ -138,23 +140,20 @@ class SelectInputWithList extends React.PureComponent {
             onChange,
             value,
             options,
-            disabled,
         } = this.props;
 
-        if (!disabled) {
-            const index = value.indexOf(key);
-            const newValue = [...value];
+        const index = value.indexOf(key);
+        const newValue = [...value];
 
-            if (index !== -1) {
-                newValue.splice(index, 1);
-            }
+        if (index !== -1) {
+            newValue.splice(index, 1);
+        }
 
-            const objectValues = this.getObjectFromValue(options, newValue);
-            this.setState({ objectValues });
+        const objectValues = this.getObjectFromValue(options, newValue);
+        this.setState({ objectValues });
```

Comment:
this setState will also be called at componentWillRecieve props so this should not be done before onChange

Reply:
done